### PR TITLE
Parameterize Big IQ loginProviderName

### PR DESCRIPTION
**Pull request by msummers-nr**
#### Description of the changes

The Big IQ loginProviderName is hard coded in the program to "tmos", this should be a configuration parameter.

### Author

- [ ] the PR should focus on a single subject. Change only relevant files to the problem you’re working on
Clean and format the code
- [ ] add unit tests for your changes and make sure all unit tests are passing
- [ ] add a risk label after carefully considering the "blast radius" of your changes
- [ ] describe the _intent_ of your changes in the description. don't just rewrite your code in prose
- [ ] assign at least one reviewer
@camdencheek 
- [ ] address the feedback 

### Reviewer

- [ ] review code for readability
- [ ] verify that high risk behavior changes are well tested
- [ ] check license for any new external dependency
- [ ] ask questions about anything that isn't clear and obvious
- [ ] approve the PR when you consider it's good to merge

### CODE REVIEW

File: src/arguments/arguments.go [hunk 1]
```diff
@@ -11,14 +11,17 @@ import (
...
-	PartitionFilter string `default:"[\"Common\"]" help:"JSON array of partitions to collect"`
+	Hostname          string `default:"localhost" help:"The hostname or IP of the F5 BIG IP device to monitor."`
+	Port              int    `default:"443" help:"The port of the iControl API to connect to."`
+	Username          string `default:"" help:"The username to connect to the F5 API with."`
+	Password          string `default:"" help:"The password to connect to the F5 API with."`
+	Timeout           int    `default:"30" help:"The number of seconds to wait before a request times out."`
+	CABundleFile      string `default:"" help:"Alternative Certificate Authority bundle file"`
+	CABundleDir       string `default:"" help:"Alternative Certificate Authority bundle directory"`
+	PartitionFilter   string `default:"[\"Common\"]" help:"JSON array of partitions to collect"`
+	AuthHost          string `default:"" help:"Hostname or IP address of the Big IQ auth token provider."`
```
Comment (camdencheek) [inline src/arguments/arguments.go:22]: It appears that `AuthHost` and `AuthPort` were added without being used anywhere. Are these required for alternative login providers?

Comment (msummers-nr) [reply src/arguments/arguments.go:22]: Odd, I don't recall adding those two lines. Let me ask the F5 tech rep if it's needed.

Comment (msummers-nr) [reply src/arguments/arguments.go:22]: Our F5 contact says "Yes", we should parameterize AuthHost & AuthPort as this is not necessarily the iControl host/port.

Tests are failing with that change, once I get them to pass I'll resubmit.

Comment (camdencheek) [reply src/arguments/arguments.go:22]: Great, sounds good. Thanks!

File: src/arguments/arguments.go [hunk 2]
```diff
@@ -19,7 +19,7 @@ type ArgumentList struct {
 	CABundleFile      string `default:"" help:"Alternative Certificate Authority bundle file"`
 	CABundleDir       string `default:"" help:"Alternative Certificate Authority bundle directory"`
 	PartitionFilter   string `default:"[\"Common\"]" help:"JSON array of partitions to collect"`
-	AuthHost          string `default:"" help:"Hostname or IP address of the Big IQ auth token provider."`
+	AuthHost          string `default:"localhost" help:"Hostname or IP address of the Big IQ auth token provider."`
```
Comment (camdencheek) [inline src/arguments/arguments.go:22]: I think having this default to `localhost` is a breaking change. Most likely, we'd want this to default to whatever `Hostname` is set to.

**Comment by msummers-nr** [comment]
Closing so I can resubmit with updates

**Pull request closed without merging**